code/datums/gamemode/role/weeaboo.dm

```diff
@@ -30,3 +30,158 @@
+						target_ground = T
+						break
+				if(!target_ground && bright_grounds.len)
+					target_ground = pick(bright_grounds)
+				if(target_ground)
+					var/datum/effect/effect/system/smoke_spread/smoke = new /datum/effect/effect/system/smoke_spread()
+					smoke.set_up(3, 0, get_turf(H))
+					smoke.start()
+					//recharge = 10 SECONDS
+					H.say("Substitution no jutsu!")
+					forceMove(get_turf(H))
+					H.forceMove(target_ground)
+					return TRUE
+				else
+					to_chat(H,"<span class='warning'>There wasn't full to teleport to!</span>")
+
+	return FALSE
+
+/*/obj/item/weapon/dakimakura/examine(mob/user)
+	..()
+	if(ishuman(user))
+		var/mob/living/carbon/human/H = user
+		if(H.mind.GetRole(WEEABOO))
+			to_chat(H,"<span class='info'>It will be ready to substitute in [recharge/10] seconds.")*/
+
+/obj/item/weapon/dakimakura/attack_hand(mob/user)
+	if(isturf(loc))
+		if(ishuman(user))
+			var/mob/living/carbon/human/H = user
+			if(H.mind.GetRole(WEEABOO))
+				..()
+			else
+				to_chat(H,"<span class='warning'>You really, really don't want to pick that up.</span>")
+	else
+		..() //You can handle it normally if it's not on the ground
+
+//The mighty power glove. Not to be confused with engineering power gloves, of course.
+/obj/item/clothing/gloves/nentendiepower
+	name = "Nen/tendie power glove"
+	desc = "Combines the power of 'Nen' (sense) with grease-resistant properties so you can still eat your tendies."
+	siemens_coefficient = 0
+	max_heat_protection_temperature = GLOVES_MAX_HEAT_PROTECTION_TEMPERATURE
+	heat_conductivity = INS_GLOVES_HEAT_CONDUCTIVITY
+	pressure_resistance = 200 * ONE_ATMOSPHERE
+	var/cooldown = 0
+
+/obj/item/clothing/gloves/nentendiepower/examine(mob/user)
+	..()
+	if(ishuman(user))
+		var/mob/living/carbon/human/H = user
+		if(H.mind.GetRole(WEEABOO))
+			to_chat(H,"<span class='info'>It will be ready to blast an APC in [max(0,cooldown-world.time)/10] seconds.")
+
+/obj/item/clothing/gloves/nentendiepower/Touch(atom/A, mob/living/user, prox)
+	if(world.time > cooldown)
+		if(ishuman(user))
+			var/mob/living/carbon/human/H = user
+			if(H.mind.GetRole(WEEABOO))
+				if(istype(A,/obj/machinery/power/apc))
+					var/obj/machinery/power/apc/APC = A
+					APC.cell.charge = 0
+					var/turf/simulated/floor/T = get_turf(APC)
+					if(istype(T))
+						T.break_tile()
+					APC.terminal.Destroy()
+					cooldown = world.time + 10 SECONDS
+	else
+		..()
+
+/obj/item/mounted/poster/anime
+	name = "rolled-up anime poster"
+	build_time = 0
+	path = /obj/structure/sign/poster/anime
+
+/obj/item/mounted/poster/anime/do_build(turf/on_wall, mob/user)
+	var/obj/structure/sign/poster/P = ..()
+	if(ishuman(user))
+		var/mob/living/carbon/human/H = user
+		if(H.mind.GetRole(WEEABOO) && P)
+			user.forceMove(P)
+	return P
+
+obj/structure/sign/poster/anime/New()
+	..()
+	name = "anime poster"
+	desc = "It's everybody's favorite anime."
+	icon_state = pick("animeposter1")
+
+obj/structure/sign/poster/anime/relaymove(mob/user as mob)
```

Comment:
firstly you're missing some slashes here and in the above New()

Secondly, this doesn't seem to destroy the poster should the weeb leave it. I fear for the hallways that are covered in anime posters.